pennylane/wires.py

```diff
@@ -26,36 +26,32 @@ class WireError(Exception):
 def _process(wires):
     """Converts the input to a tuple of numbers or strings."""
 
-    if isinstance(wires, Wires):
-        # if input is already a Wires object, just return its wire tuple
-        return wires.labels
-
     if isinstance(wires, (Number, str)):
         # interpret as a single wire
         return (wires,)
 
-    if hasattr(wires, "shape") and wires.shape == tuple():
+    if isinstance(wires, Wires):
+        # if input is already a Wires object, just return its wire tuple
+        return wires.labels
+
+    if getattr(wires, "shape", None) == tuple():
         # Scalar NumPy array
         return (wires.item(),)
 
     if isinstance(wires, Iterable):
-        # wires is an iterable, presumably of Wires, Number or scalar NumPy array instances
-        if all(isinstance(w, Wires) for w in wires):
-            # if the elements are themselves Wires objects, merge them to a new one
-            merged = tuple(w for wires_ in wires for w in wires_.tolist())
-        elif all(
-            isinstance(w, (str, Number)) or (getattr(w, "shape", None) == tuple()) for w in wires
-        ):
-            # if the elements are strings or numbers, turn iterable into tuple
-            merged = tuple([w.item() if isinstance(w, np.ndarray) else w for w in wires])
-        else:
-            merged = None
-
-        if merged is not None:
-            # check that all wires are unique
-            if len(set(merged)) != len(merged):
-                raise WireError("Wires must be unique; got {}.".format(merged))
-            return merged
+
+        tuple_of_wires = tuple(wires)
+        try:  # We need the set for the uniqueness check, so we can use it for hashability check
+            set_of_wires = set(wires)
+        except TypeError as e:
+            # Make sure it really was a hashability issue
+            if str(e).startswith("unhashable"):
+                raise WireError("Wires must be hashable; got {}.".format(wires)) from e
```

Comment:
I think we should reraise `e` if it wasn't this error.

Reply:
Actually, there was a dedicated error raised previously. But then codecov complained that that was not tested and I tried to find a test case raising this. Did not find one, maybe you can add it if you have an idea how to break this.